corehq/apps/users/models.py

```diff
@@ -751,19 +751,15 @@ def merge(self, other):
         if other.last_request <= self.last_request:
             return
 
-        for key, prop in self.properties().items():
+        for key, prop in other.properties().items():
             new_val = getattr(other, key)
-            if new_val:
+            if new_val is not None:
                 old_val = getattr(self, key)
-                if not old_val:
-                    setattr(self, key, new_val)
-                    continue
 
                 prop_is_date = isinstance(prop, DateTimeProperty)
-                if prop_is_date and new_val > old_val:
-                    setattr(self, key, new_val)
-                elif not prop_is_date and old_val != new_val:
-                    setattr(self, key, new_val)
```

Comment:
What about this logic?  Do you know why the old version skipped setting the val if it was the same?  Seems useless, but I wonder if there's some spy logic that only saves this if an attr has been set.  If so, this change would result in extra writes.  Seems like a bit of a stretch, just wanted to probe into the rationale behind the old version of the code.

Reply:
I don't think there's anything like that going on.

Reply:
Thanks for keeping an eye out for these edge cases. `DeviceAppMeta` only inherits from `DocumentSchema`, which, as far as I'm aware, doesn't do anything special, and also...is a couch model, which we'll likely migrate away from. Without a comment specifically calling out that we're doing something for performance reasons, I think it's worth it to simplify the logic